src/main/java/com/exasol/adapter/dialects/sybase/SybaseIdentifier.java

```diff
@@ -0,0 +1,76 @@
+package com.exasol.adapter.dialects.sybase;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.exasol.db.Identifier;
+
+/**
+ * Represents an identifier in the Sybase database.
+ */
+public class SybaseIdentifier implements Identifier {
+    private static final Set<Character> ALLOWED_CHARS = Set.of(' ', '_', '@', '#', '$');
+    private final String id;
+
+    private SybaseIdentifier(final String id) {
+        this.id = id;
+    }
+
+    /**
+     * Get the quoted identifier as a {@link String}.
+     *
+     * @return quoted identifier
+     */
+    @Override
+    public String quote() {
+        return "[" + this.id + "]";
```

Comment:
I can't find the `[]` in the documentation. And the ALLOWED_CHARS seems also different and I'm not sure whether it applies to delimited identifiers.

Reply:
There is some mentions here: http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1600/doc/html/san1393050529478.html

Reply:
I can't find anything about `[]` there. Please notice that the URL doesn't change when clicking some pages. The page http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1600/doc/html/san1393050531431.html mentions `"`, but that depends on the configuration, so we should try to avoid it. According to http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1600/doc/html/san1393050535150.html and http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.infocenter.dc36271.1600/doc/html/san1393050535962.html the validity even depends on the configured character set.
Might be the best idea to fix the ALLOWED_CHARS (spaces, currency symbols) and make our checks very restrictive, e.g. exclude Unicode, and then use unquoted names. But that might also be too restrictive.